### DISCUSSION TITLE
PageRank recommendations to account for popularity bias

### DISCUSSION THREAD

**Discussion by levrone1987** [sknetwork-team/scikit-network#541] [Ideas]
I recently stumbled upon this [post](https://www.inovex.de/de/blog/fairness-in-recommender-systems-how-to-reduce-the-popularity-bias/)

The author describes a PageRank adaptation where the transition depends on the perceived item popularity. Is there a way to utilize PageRank for personalized recommendations, as described [here](https://scikit-network.readthedocs.io/en/latest/use_cases/recommendation.html), but also make the transitions depend on node degrees (or a list of perceived inversed popularities)? How would you approach this implementation?

Thanks!

**Comment by tbonald** [comment]
Hi,
I think the best way would be to pre-process the item-user biadjacency matrix (e.g., by multiplying each row by the corresponding perceived item popularity), then to run PageRank on this new matrix.

**Reply by levrone1987** [reply]
Thanks tbonald! Since the probability of the transitions would be proportional to the multiplication factor, and we would like to lower the impact of high-degree nodes, the multiplication factor should be inverse of the item popularity (degree).
